### PR TITLE
V1.0.4

### PR DESCRIPTION
**Pull request by anvesh-cohesity**
- Native agent installation support (qualified for RedHat, CentOS, Ubuntu)
- Ability to install the agent using installer from custom download location
- Ability to include and exclude vm’s, resource pools, folders in an existing or new VMware protection job
- Ability to specify backup timestamp for the file restore operation
- Updated documentation
- Additional demo examples

### CODE REVIEW

File: docs/examples/demos/agents.md [hunk 1]
```diff
@@ -72,14 +72,14 @@ The combined source file for these two playbooks is located at the root of the r
...
-Here is an example playbook that installs the Cohesity agent on all `linux` hosts. (Remember to change it to suit your environment.)
+Here is an example playbook that uninstalls and installs the Cohesity agent on all `linux` hosts. (Remember to change it to suit your environment.)
 > **Notes:**
   - Before using these example playbooks, refer to the [Setup](../../setup.md) and [How to Use](../../how-to-use.md) sections of this guide.
   - When using the default download location, the Cohesity agent installer is placed in `/tmp/<temp-dir>`.  If your environment prevents the use of `/tmp` with a `noexec` option, then you must set an alternate download location.
 
 You can create a file called `deploy-cohesity-agent-linux.yml`, add the contents from the sample playbook, and then run the playbook using `ansible-playbook`:
   ```
-  ansible-playbook -i <inventory_file> deploy-cohesity-agent-linux.yml -e "username=admin password=admin"
+  ansible-playbook -i <inventory_file> deploy-cohesity-agent-linux.yml -e "username=abc password=abc"
```
Comment (pyashish) [inline docs/examples/demos/agents.md:82]: use 
username=<username>
password=<password>

File: docs/modules/cohesity_job.md [hunk 2]
```diff
@@ -111,11 +111,11 @@ This Ansible Module registers, removes, starts, and stops the Cohesity Protectio
     state: present
     name: myvcenter
     environment: VMware
-    sources:
+    protection_sources:
```
Comment (pyashish) [inline docs/modules/cohesity_job.md:114]: wont this break backward compatibility ?

File: docs/modules/cohesity_job.md [hunk 3]
```diff
@@ -111,11 +111,11 @@ This Ansible Module registers, removes, starts, and stops the Cohesity Protectio
...
       - endpoint: myvcenter.domain.lab
     protection_policy: Gold
     storage_domain: Default
-    include_vms:
+    include:
```
Comment (pyashish) [inline docs/modules/cohesity_job.md:118]: similar comment as above

File: examples/demos/add_job_linux.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@
           name: cohesity.cohesity_ansible_role
           tasks_from: job
         vars:
-          cohesity_server: 10.2.145.27
+          cohesity_server: 10.22.149.28
```
Comment (pyashish) [inline examples/demos/add_job_linux.yml:13]: remove these IPs  ??

File: library/cohesity_job.py [hunk 5]
```diff
@@ -348,6 +348,53 @@ def create_paths_parameter(module, update_source_ids):
...
+        objects = nodes.pop()
+        for node in objects:
+            if 'nodes' in node:
+                nodes.append(node['nodes'])
+            if ('protectionSource' in node) and (node['protectionSource']['name'] in vm_names):
+                ids.append(node['protectionSource']['id'])
+    return list(set(ids))
+
+
+def get_vmware_ids(module, job_meta_data, job_details, vm_names):
```
Comment (pyashish) [inline library/cohesity_job.py:370]: going forward any new functionalities added use the python sdk.

File: library/cohesity_restore_vm.py [hunk 6]
```diff
@@ -379,7 +379,7 @@ def wait_restore_complete(module, self):
         while attempts < wait_counter:
 
             response = open_url(url=uri, headers=headers,
-                                validate_certs=validate_certs)
+                                validate_certs=validate_certs, timeout=120)
```
Comment (pyashish) [inline library/cohesity_restore_vm.py]: Good catch using the timeout. Can you make this a global constant ?

Comment (anvesh-cohesity) [reply library/cohesity_restore_vm.py]: Ok

**Pull request merged**